packages/rocketchat-livechat/server/methods/registerGuest.js

```diff
@@ -7,9 +7,6 @@ Meteor.methods({
 			department
 		});
 
-		// update visited page history to not expire
-		RocketChat.models.LivechatPageVisited.keepHistoryForToken(token);
```

Comment:
we should keep this behavior. the idea is that page history of unregistered visitors can be deleted (expired), because we track visitor's navigation from the very first interaction.. but if that visitor sends a message then we want to keep his page history "forever"

Reply:
There is a problem here.. I tried to keep this behavior, but currently we are sending and storing the navigation history as a message and we have a relation between message and room.
The point is that the method `RocketChat.models.Messages.createWithTypeRoomIdMessageAndUser` calls another method `RocketChat.models.Rooms.incMsgCountById(room._id, 1)` and a valid room is required.
So, because of this we can't store messages without a room using this method.
If you could give me an orientation about this it would be nice.

Reply:
I'll give this another look, but since we merge https://github.com/RocketChat/Rocket.Chat/pull/9903 we will not call `incMsgCountById` anymore.. so we might be able to keep this behavior.. 😉